indexer/services/roundtable/src/tasks/update-affiliate-info.ts

```diff
@@ -0,0 +1,60 @@
+import { logger } from '@dydxprotocol-indexer/base';
+import {
+  PersistentCacheTable,
+  AffiliateInfoTable,
+  PersistentCacheKeys,
+  PersistentCacheFromDatabase,
+  BlockFromDatabase,
+  BlockTable,
+  Transaction,
+} from '@dydxprotocol-indexer/postgres';
+import { DateTime } from 'luxon';
+
+const defaultLastUpdateTime: string = '2024-09-16T00:00:00Z';
+
+/**
+ * Update the affiliate info for all affiliate addresses.
+ */
+export default async function runTask(): Promise<void> {
+  // Wrap getting cache, updating info, and setting cache in one transaction so that persistent
+  // cache and affilitate info table are in sync.
+  const txId: number = await Transaction.start();
+  try {
+    const latestBlock: BlockFromDatabase = await BlockTable.getLatest();
+    if (latestBlock.time === null) {
+      throw Error('Failed to get latest block time');
+    }
+    const persistentCacheEntry: PersistentCacheFromDatabase | undefined = await PersistentCacheTable
+      .findById(PersistentCacheKeys.AFFILIATE_INFO_UPDATE_TIME, { txId });
+    if (!persistentCacheEntry) {
+      logger.info({
+        at: 'update-affiliate-info#runTask',
+        message: `No previous ${PersistentCacheKeys.AFFILIATE_INFO_UPDATE_TIME} found in persistent cache table. Will use default value: ${defaultLastUpdateTime}`,
+      });
+    }
+    const windowStartTime: DateTime = DateTime.fromISO(persistentCacheEntry
+      ? persistentCacheEntry.value
+      : defaultLastUpdateTime);
+
+    let windowEndTime = DateTime.fromISO(latestBlock.time);
+    // During backfilling, we process one day at a time to reduce roundtable runtime.
+    if (windowEndTime > windowStartTime.plus({ days: 1 })) {
+      windowEndTime = windowStartTime.plus({ days: 1 });
+    }
+
+    await AffiliateInfoTable.updateInfo(windowStartTime.toISO(), windowEndTime.toISO(), { txId });
```

Comment:
nit: let's add a log to print out window start and end time

Reply:
sure